app/uk/gov/hmrc/helptosave/actors/ThresholdMongoProxy.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.actors
+
+import akka.pattern.pipe
+import akka.actor.{Actor, Props}
+import cats.syntax.either._
+import uk.gov.hmrc.helptosave.actors.ThresholdMongoProxy._
+import uk.gov.hmrc.helptosave.repo.ThresholdStore
+
+class ThresholdMongoProxy(thresholdStore: ThresholdStore) extends Actor {
```

Comment:
@AndyHWChung why do we need to do this ? 
we just store threshold in the mongo store and thats it - we don't care even if there are issues during storing to mongo because the next call after 24h will make another update ?

Reply:
When the backend starts we won't have an in-memory value to use and in this state no eligibility checks can be done. We need to retrieve a value as quickly as possible and if DES is down we use mongo as a backup. In this scenario, we need to make sure that the value in mongo is as up-to-date as possible.
The chances that mongo has the wrong value are very slim: it'd have to be the day when the change is made and the write to mongo failed and then we redeployed the application before the updated value was stored to mongo. But since this chance still exists, it is safer to ensure we definitely write the value to mongo.

Reply:
in reality, we're not expecting that writes to mongo are going to fail for a long period of time - but at least if it is we're prepared

Reply:
if the write to mongo fails and we don't retry there is still the danger that the value read from mongo on the next application startup is going to be wrong. The retries are only done when there is an error writing to mongo - the retries are stopped once a successful write occurs

Reply:
`the value read from mongo on the next application startup is going to be wrong` -  yes, this is actually an issue only if the update that failed tried to update with a new/updated threshold value which is a rare case  given that how frequently the ITMP threshold value changes...
any way there is no harm in keeping the re-try logic in the codebase...